### PR TITLE
Intermediate: Pass io tests

### PR DESCRIPTION
**Pull request by znicholls**
Intermediate step towards solving #197. Fails CI as only the io tests have passed here.

@rgieseke the typing stuff sven has done in openscm mean that the code is now Python3.5 incompatible, are we ok with that?

### CODE REVIEW

**Comment by rgieseke** [comment]
> @rgieseke the typing stuff sven has done in openscm mean that the code is now Python3.5 incompatible, are we ok with that?

Not sure what exactly the problem is, i thought `typing` was included in 3.5 - is a specific new feature the problem?

**Comment by znicholls** [comment]
> Not sure what exactly the problem is, i thought `typing` was included in 3.5 - is a specific new feature the problem?

It appears not, from https://travis-ci.org/openclimatedata/pymagicc/jobs/484226040

```
E     File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/openscm/core.py", line 28
E       _world: _Region
E             ^
E   SyntaxError: invalid syntax
```

whilst https://travis-ci.org/openclimatedata/pymagicc/jobs/484226041 raises no such syntax error

**Comment by rgieseke** [comment]
Hm, it seems this syntax is 3.6+ only.

Let's drop 3.5 support! If someone *needs* to run this on a server they can get a more recent Python with Conda.

File: pymagicc/io.py [hunk 1]
```diff
@@ -242,20 +246,46 @@ def _convert_data_block_and_headers_to_df(self, stream, column_headers):
...
+            "climate_model",
+        ]:
             df[categorical_column] = df[categorical_column].astype(str)
 
         return df
 
     def _get_columns_multiindex_from_column_headers(self, ch):
+        l = len(ch["variables"])
+        ch.setdefault("climate_models", ["MAGICC"] * l)
+        ch.setdefault("models", ["tbc"] * l)
```
Comment (rgieseke) [inline pymagicc/io.py]: Is "tbc" justs a placeholder or does it need to be changed later?

**Comment by znicholls** [comment]
placeholder, changed to "unspecified" in #201 or #202

**Pull request merged**